pkg/config/constants/constants.go

```diff
@@ -172,6 +172,11 @@ const (
 	RemoteGatewayClassName   = "istio-remote"
 	WaypointGatewayClassName = "istio-waypoint"
 
+	// GatewayAddressTypeOverride allows overriding the address type for a Gateway
```

Comment:
I think we can get away with always writing both actually.. then just use the IP in our ambient controller

but I don't mind this for now as well

Reply:
Changed it to do that.

Reply:
https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/50132/integ-ds_istio/1773261172389187584

Actually, this breaks conformance tests.